ontimize/components/form/o-form.component.ts

```diff
@@ -788,10 +821,20 @@ export class OFormComponent implements OnInit, OnDestroy, CanComponentDeactivate
       self.postCorrectInsert(resp);
       self.formCache.setCacheSnapshot();
       self.markFormLayoutManagerToUpdate();
-      if (self.stayInRecordAfterInsert) {
+      if (self.stayInRecordAfterInsert || self.afterInsertMode === 'detail') {
         self._stayInRecordAfterInsert(resp);
+        if (self.afterInsertMode === 'new') {
```

Comment:
Does this make sense here? "afterInsertMode" is checked just before this line and with other value

Reply:
I did this because maybe the user adds the both attributes at the same time but if you think this won´t occur I can remove it.